partner_firstname/models/res_partner.py

```diff
@@ -125,7 +141,9 @@ def _get_whitespace_cleaned_name(self, name):
 
         Removes leading, trailing and duplicated whitespace.
         """
-        return u" ".join(name.split(None)) if name else name
+        name = name.strip() if name else ''
+        name = re.sub(r'([ ]) +', r' ', name)
+        return re.sub(r' *[,] *', r',', name)
```

Comment:
I think this is more complicated than the previous version with no benefit. `u" ".join(name.split(None))` [already unifies all whitespace](https://docs.python.org/2/library/stdtypes.html#str.split). The only remaining corner case would be a space between the previous word of the comma and the coma, like in `lastname , firstname`, but that can be fixed with [`name.replace(" ,", ",")`](https://docs.python.org/2/library/stdtypes.html#str.replace) without need for regular expressions.

Also this function should return `False` in case of empty name. This way you force other methods to check that name has a value before doing any str-related stuff, and otherwise database could end up storing `""` instead of `False`, which will lead to badness.